tls/extensions/s2n_client_server_name.c

```diff
@@ -61,23 +61,21 @@ static int s2n_client_server_name_send(struct s2n_connection *conn, struct s2n_s
 /* Read the extension up to the first item in ServerNameList. Store the first entry's length in server_name_len.
  * For now s2n ignores all subsequent items in ServerNameList.
  */
-static int s2n_client_server_name_check(struct s2n_connection *conn, struct s2n_stuffer *extension, uint16_t *server_name_len)
+S2N_RESULT s2n_validate_server_name_length(struct s2n_stuffer *extension, uint16_t *server_name_len)
 {
```

Comment:
This seems like a really deceptive function name. When I first read it I was like "oh no, did we remove the host name type check?". It's validating more than just the length.

It could also probably output a blob of the server name. Then the getter would just copy that blob.

Reply:
I don't know why; I'm not following your suggestion. Are you saying that this function should return a pointer to where the server name begins?

Reply:
Yes. And if you use a blob (via s2n_blob_ini), that will also include the length.